app/src/main/java/com/pps/globant/fittracker/mvp/presenter/LoginPresenter.java

```diff
@@ -0,0 +1,166 @@
+import com.pps.globant.fittracker.mvp.model.LoginModel;
+import com.pps.globant.fittracker.mvp.view.LoginView;
+import com.squareup.otto.Subscribe;
+
+import com.pps.globant.fittracker.R;
+import com.pps.globant.fittracker.utils.FacebookLoginProvider;
+
+import android.content.res.Resources;
+import android.util.Log;
+
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.widget.Toast;
+
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+
+import static com.pps.globant.fittracker.mvp.view.LoginView.*;
+import static com.pps.globant.fittracker.mvp.model.LoginModel.*;
+
+public class LoginPresenter {
+
+    private final LoginModel model;
+    private final LoginView view;
+
+    //Google declarations-----------------------------------------------------------------------------------------------------------
+    /*googleServiceClientId is a key obtained from https://developers.google.com/identity/sign-in/android/start-integrating
+    To get it, it requires an unique SHA1 key, so if you want to recompile the app in another pc, you'll need to create a new key.*/
+    private static final String GOOGLE_SERVICE_CLIENT_ID = "268582315609-j419amnke1b8djg935oq1ncd08e78lam.apps.googleusercontent.com";
+    private static final String GOOGLE_SIGN_IN_ERROR_TAG = "Sign In Error";
+    private static final String GOOGLE_SIGNED_OUT_MESSAGE = "Signed out from google";
+    private static final String GOOGLE_SIGN_IN_ERROR_MESSAGE ="handleSignInResult:error";
+    private static final String EMPTY_STRING ="";
+    private static final int RC_GET_TOKEN = 9002;
+
+
+    private GoogleSignInOptions gso;
+    private GoogleSignInClient mGoogleSignInClient;
+    //-------------------------------------------------------------------------------------------------------------------------------
+
+    public LoginPresenter(LoginModel model, LoginView view) {
+        this.model = model;
+        this.view = view;
+
+        // Configure sign-in to request the user's ID, email address, token and basic
+        // profile. ID and basic profile are included in DEFAULT_SIGN_IN.
+        gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(GOOGLE_SERVICE_CLIENT_ID)
+                .requestEmail()
+                .build();
+
+        // Build a GoogleSignInClient with the options specified by gso.
+        if (view.getActivity() != null)
+            mGoogleSignInClient = GoogleSignIn.getClient(view.getActivity(), gso);
+
+        // [START on_start_sign_in]
+        // Check for existing Google Sign In account, if the user is already signed in
+        // the GoogleSignInAccount will be non-null.
+        if (view.getActivity() != null){
```

Comment:
You are checking the same thing 2 times. I think that you should unify these

Reply:
@nacho-globant  it is necessary to check for null since the presenter? the presenter is initialized and destroyed along with the activity